packages/story-editor/src/components/library/panes/shopping/productButton.js

```diff
@@ -100,28 +121,32 @@ function ProductButton({ product, onClick, onFocus, isOnPage }) {
     product?.productTitle
   );
 
+  const disabled = !isOnPage && !canAddMore;
```

Comment:
```suggestion
  const disabled = (!isOnPage && !canAddMore) || !hasImage;
```

Makes the `hasImage && !disabled` and `!hasImage || disabled` checks below easier to maintain as there's only 1 variable. WDYT?

Reply:
I don't think that is going to work. There are cases were we need to know disabled and HasImage separately. I also find that a little less readable.